crates/bitwarden/src/auth/password/strength.rs

```diff
@@ -0,0 +1,99 @@
+use std::collections::HashSet;
+
+use zxcvbn::zxcvbn;
+
+const GLOBAL_INPUTS: [&str; 3] = ["bitwarden", "BW", "Bitwarden"];
+
+pub(crate) fn password_strength(
+    password: String,
+    email: String,
+    additional_inputs: Vec<String>,
+) -> u8 {
+    let email_parts = email_to_user_inputs(&email);
+    let email_ref: Vec<_> = email_parts.iter().map(String::as_str).collect();
+
+    let mut additional_inputs: HashSet<_> = additional_inputs.iter().map(String::as_str).collect();
```

Comment:
zxcvbn seems to be lowercasing the additional_inputs and adding them as keys to a hashmap, so we wouldn't need to use a set here, which would simplify the code a bit.

Reply:
I'm slightly hesitant to rely on internal behaviour.